src/system/SystemPoolHeap.h

```diff
@@ -0,0 +1,105 @@
+/*
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Copyright (c) 2016-2017 Nest Labs, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/core/CHIPConfig.h>
+
+#if CHIP_SYSTEM_CONFIG_POOL_USE_HEAP
```

Comment:
Could we refactor code to clean out "#if CHIP_SYSTEM_CONFIG_POOL_USE_HEAP" in SystemPoolHeap.h ? since this file only contains heap based implementation. The #if check should be made in SystemPool.h

Reply:
I'd prefer this way rather that 

```
#if CHIP_SYSTEM_CONFIG_POOL_USE_HEAP
#include <Impl_A>
#else
#include <Impl_B>
#endif
```

This is more robust, because people may mistakenly include wrong headers.